### PR TITLE
Add optional SELinux support to RHEL clusters

### PR DESCRIPTION
**Pull request by jsafrane**
Add `cluster.containerd.selinuxEnabled` field that enables SELinux support in containerd on RHEL8/9 (and perhaps other distros). In the end, it adds `enable_selinux=true` in containerd config file.

Unless specified in a Kubernetes Pod, the container runtime then allocates a unique random SELinux label, for example `system_u:system_r:container_t:s0:c152,c241` and runs the Pod containers with that label. If a container process then escapes its containment (which would be a serious CVE in the container runtime), SELinux will prevent the rogue process accessing files of other Pods (because they will have different `cXXX,cYYY` numbers) and on the host.

There is a glitch though - with SELinux enabled in the container runtime, unprivileged containers cannot access anything on the host. Since kops runs all Kubernetes components and etcd as static Pods, these Pods can't e.g. read `/etc` and write to `/var/log/`. Therefore I added `securityOptions: seLinuxOptions: type: spc_t` to most static Pods and all addons that use HostPath volumes. Containers running with `spc_t` (abrev. of "super privileged containers") can access any file on the host. (Regular UID/GID checks are still in effect)

With the changes here, I am able to run a cluster with cillium on RHEL8+9. I did not check other distros nor network plugins.

WIP:
~* needs https://github.com/kubernetes/kops/pull/15496 merged first~

### CODE REVIEW

**Review by hakman — CHANGES_REQUESTED** [review]
@jsafrane I don't think we need feature flag for this. We already had it enabled for Docker in the past. By default it would be disabled anyway.

I would suggest adding a new template function to check for containerd selinux:
```
	dest["ContainerdSELinuxEnabled"] = func() bool {
		if cluster.Spec.Containerd != nil  {
			return cluster.Spec.Containerd.SELinuxEnabled
		}
		return false
	}
```

File: pkg/apis/kops/containerdconfig.go [hunk 1]
```diff
@@ -43,6 +43,8 @@ type ContainerdConfig struct {
 	NvidiaGPU *NvidiaGPUConfig `json:"nvidiaGPU,omitempty"`
 	// Runc configures the runc runtime.
 	Runc *Runc `json:"runc,omitempty"`
+	// SelinuxEnabled enables SELinux support
+	SeLinuxEnabled *bool `json:"selinuxEnabled,omitempty"`
```
[the remark this answers is not in the document]
Comment (jsafrane) [reply pkg/apis/kops/containerdconfig.go]: No special reason for pointer, changed to bool.

**Comment by hakman** [comment]
@jsafrane Probably you need to run also `hack/update-expected.sh`. Otherwise LGTM!

**Review by hakman — APPROVED** [review]

**Pull request merged**